packages/web-api/test/types/webclient-paginate-types.test-d.ts

```diff
@@ -0,0 +1,57 @@
+/// <reference lib="esnext.asynciterable" />
+
+import { expectType, expectError } from 'tsd';
+import { WebClient, WebAPICallResult } from '../../';
+
+const web = new WebClient();
+
+/* Testing the return type of WebClient#paginate() */
+
+expectType<AsyncIterable<WebAPICallResult>>(web.paginate('conversations.list'));
+
+expectType<AsyncIterable<WebAPICallResult>>(web.paginate('conversations.list', {}));
+
+expectType<Promise<void>>(web.paginate('conversations.list', {}, () => false));
+
+expectError(web.paginate('conversations.list', {}, () => 7));
```

Comment:
nit: It may be also fine if we could make the validation here more specific by having `T` in `expectError<T>(value)`.

Reply:
I think the two are subtly different as `expectError<T>` checks a value rather than function arguments. In all honesty, I don't know what the difference between `expectError<T>` and `expectNotAssignable<T>` is other than the error message.

That said, I don't see how `expectError<T>` would for this case?

Reply:
Ah never mind! Somehow, I was just a bit confused at somewhere between type tests and runtime tests...